webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/DatasetArray.scala

```diff
@@ -116,37 +120,57 @@ class DatasetArray(vaultPath: VaultPath,
       asBytes <- BytesConverter.toByteArray(typedMultiArray, header.resolvedDataType, ByteOrder.LITTLE_ENDIAN)
     } yield asBytes
 
+  private def printAsInner(values: Array[Int], flip: Boolean = false): String = {
+    val axisNames = fullAxisOrder.axes.map(_.name)
+    val axisNamesFlippedIfNeeded = if (flip) axisNames.reverse else axisNames
+    val raw = axisNamesFlippedIfNeeded
+      .zip(values)
+      .map { tuple =>
+        f"${tuple._1}=${tuple._2}"
+      }
+      .mkString(",")
+    f"inner($raw)"
+  }
+
+  private def printAsOuterF(values: Array[Int]): String = {
+    val axisNamesFOrder = fullAxisOrder.axesWk.map(_.name).reverse
+    val raw = axisNamesFOrder
+      .zip(values)
+      .map { tuple =>
+        f"${tuple._1}=${tuple._2}"
+      }
+      .mkString(",")
+    f"outer($raw)"
+  }
+
   // Read from array. Note that shape and offset should be passed in XYZ order, left-padded with 0 and 1 respectively.
   // This function will internally adapt to the array's axis order so that XYZ data in fortran-order is returned.
-
   private def readAsFortranOrder(shape: Array[Int], offset: Array[Int])(
       implicit ec: ExecutionContext): Fox[MultiArray] = {
     val totalOffset: Array[Int] = offset.zip(header.voxelOffset).map { case (o, v) => o - v }.padTo(offset.length, 0)
-    val chunkIndices = ChunkUtils.computeChunkIndices(datasetShape.map(axisOrder.permuteIndicesReverse),
-                                                      axisOrder.permuteIndicesReverse(chunkShape),
+    val chunkIndices = ChunkUtils.computeChunkIndices(datasetShape.map(fullAxisOrder.permuteIndicesArrayToWk),
+                                                      fullAxisOrder.permuteIndicesArrayToWk(chunkShape),
                                                       shape,
                                                       totalOffset)
     if (partialCopyingIsNotNeeded(shape, totalOffset, chunkIndices)) {
       for {
         chunkIndex <- chunkIndices.headOption.toFox
```

Comment:
```suggestion
        chunkIndexInWk <- chunkIndices.headOption.toFox
```
?

Reply:
I’m hesitant, because then we would have to annotate shape, offset, chunkIndices, sourceChunk as well. It’s true that this ordering is important to understanding this function, but it is not the only thing going on here, so I’m scared the renamings would generate too much noise. Instead, I tried to make the comment more descriptive. What do you think?